src/annotator/util/emitter.js

```diff
@@ -0,0 +1,74 @@
+import { TinyEmitter } from 'tiny-emitter';
+
+/**
+ * Emitter is a communication class that implements the publisher/subscriber
+ * pattern. It allows sending and listening events through a shared EventBus.
+ * The different elements of the application can communicate with each other
+ * without being tightly coupled.
+ */
+class Emitter {
+  /**
+   * @param {TinyEmitter} emitter
+   */
+  constructor(emitter) {
+    this._emitter = emitter;
+
+    /** @type {[event: string, callback: Function][]} */
+    this._subscriptions = [];
+  }
+
+  /**
+   * Fire an event.
+   *
+   * @param {string} event
+   * @param {any[]} args
+   */
+  publish(event, ...args) {
+    this._emitter.emit(event, ...args);
+  }
+
+  /**
+   * Register an event listener.
+   *
+   * @param {string} event
+   * @param {Function} callback
+   */
+  subscribe(event, callback) {
+    this._emitter.on(event, callback);
+    this._subscriptions.push([event, callback]);
+  }
+
+  /**
+   * Remove an event listener.
+   *
+   * @param {string} event
+   * @param {Function} callback
+   */
+  unsubscribe(event, callback) {
+    this._emitter.off(event, callback);
+    this._subscriptions = this._subscriptions.filter(
+      ([subEvent, subCallback]) =>
+        subEvent !== event || subCallback !== callback
+    );
+  }
+
+  /**
+   * Remove all event listeners.
+   */
+  destroy() {
+    for (let [event, callback] of this._subscriptions) {
+      this._emitter.off(event, callback);
+    }
+    this._subscriptions = [];
+  }
+}
+
+export default class EventBus {
```

Comment:
In recent code we've started to move away from using default exports in most scenarios. The reason is that it can lead to the same type unintentionally ending up with different names in different files which can be confusing. In other words `import FooBar from './util/some-module` can silently "rename" whatever the actual name of the default export is to `FooBar`. If importing by name you end up making this explicit at least (`import { Foo as FooBar } ...`. There is an exception for Preact components to follow ecosystem conventions.

Reply:
I fully agree. Generally, I dislike default exports.